crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -431,9 +431,28 @@ impl<'this, TX: DbTxMut<'this> + DbTx<'this>> DatabaseProvider<'this, TX> {
             .map(|(id, tx)| (id, tx.into()))
             .collect::<Vec<(u64, TransactionSigned)>>();
 
-        let senders =
+        let mut senders =
             self.get_or_take::<tables::TxSenders, TAKE>(first_transaction..=last_transaction)?;
 
+        // Recover senders manually if not found in db
+        let senders_len = senders.len();
+        let transactions_len = transactions.len();
+        let missing_senders = transactions_len - senders_len;
+        let mut senders_recovered: Vec<(u64, H160)> = (first_transaction..
+            first_transaction + missing_senders as u64)
+            .zip(
+                TransactionSigned::recover_signers(
+                    transactions.iter().take(missing_senders).map(|(_, tx)| tx).collect::<Vec<_>>(),
+                    missing_senders,
+                )
+                .ok_or(BlockExecutionError::Validation(
+                    BlockValidationError::SenderRecoveryError,
+                ))?,
+            )
+            .collect();
+        senders_recovered.extend(senders.iter());
```

Comment:
let's mention in the comment that it's only possible to have missing senders at the beginning of the range, and not in the middle or in the end, so it's safe to do `senders_recovered.extend(senders.iter())`

Reply:
ok, add comment

Reply:
one last thing, let's also add a debug assert similar to this one https://github.com/paradigmxyz/reth/blob/e576c007e3d8f7e462ffb65781ade8be382a016b/crates/storage/provider/src/providers/database/provider.rs#L1917

Reply:
added, let me know if it's ok like this. Should be ok imo